src/main.ts

```diff
@@ -56,7 +57,9 @@ async function main() {
 		types: TYPES ? (require(TYPES) as RegistryTypes) : undefined,
 		/* eslint-enable @typescript-eslint/no-var-requires */
 	});
-
+	if(api.registry.chainSS58 != undefined){
```

Comment:
So this logic would actually have to be contained inside of the middleware. It would have to check if the requests are grabbing information from a past block that has a historic runtime OR if there is no `at` query parameter attached then to just use the current api's registry. This also stops us from mutating any data in polkadot-js through `{ defaults }`, and potentially having other files import that mutated dep.